app/models/manageiq/providers/vmware/infra_manager/inventory/persister.rb

```diff
@@ -61,4 +66,63 @@ def vim_class_to_collection(managed_object)
       resource_pools
     end
   end
+
+  def hosts_reconnect_block
+    lambda do |inventory_collection, inventory_objects_index, attributes_index|
+      relation = inventory_collection.model_class.where(:ems_id => nil)
+      return if relation.count <= 0
+
+      inventory_objects_index.each do |ref, obj|
+        hostname = obj.hostname
+        ipaddr   = obj.ipaddress
+
+        record = if ["localhost", "localhost.localdomain", "127.0.0.1"].include_none?(hostname, ipaddr)
+          h   = relation.where("lower(hostname) = ?", hostname.downcase).find_by(:ipaddress => ipaddr) if hostname && ipaddr
+          h ||= relation.find_by("lower(hostname) = ?", hostname.downcase)                             if hostname
+          h ||= relation.find_by(:ipaddress => ipaddr)                                                 if ipaddr
+          h ||= relation.find_by("lower(hostname) LIKE ?", "#{hostname.downcase}.%")                   if hostname
```

Comment:
This feels like a duplicate of the core method...do we still need the core method after this?

Reply:
Yeah it is a duplicate, the core method should no longer be used but I haven't gone through and figured out which `save_*_inventory` methods smartstate uses so I haven't deleted `save_hosts_inventory`.  Technically SSA shouldn't ever be reconnecting hosts though so we could update that method to just be a simple ems_id+ems_ref lookup.

Reply:
https://github.com/ManageIQ/manageiq/pull/20712